### PR TITLE
Install lso cluster + encryption via UI

### PR DESCRIPTION
**Pull request by OdedViner**
Signed-off-by: Oded Viner <oviner@redhat.com>
*Add Disks
*Install Local Storage Operator
*Install OCS Operator
*Install Storage Cluster (+ encryption)

### CODE REVIEW

File: ocs_ci/ocs/ui/deployment_ui.py [hunk 1]
```diff
@@ -243,52 +263,25 @@ def configure_encryption(self):
...
-
-            logger.info(f"kms_token: {self.kms_token}")
-            self.do_send_keys(text=self.kms_token, locator=self.dep_loc["kms_token"])
-
-    def verify_ocs_operator_succeeded(self, timeout_install=300, sleep=20):
+    def verify_operator_succeeded(
+        self, operator="OpenShift Container Storage", timeout_install=300, sleep=20
+    ):
         """
         Verify OCS Installation
```
Comment (ebondare) [inline ocs_ci/ocs/ui/deployment_ui.py:282]: Please update the description to "Verify Operator Installation"

File: ocs_ci/ocs/ui/deployment_ui.py [hunk 2]
```diff
@@ -24,112 +25,57 @@ def __init__(self, driver):
...
-    def select_kms_token(self):
-        return self.kms_token
-
-    @select_kms_token.setter
-    def select_kms_token(self, kms_token):
-        if not isinstance(kms_token, str):
-            raise ValueError("kms_token arg must be a string")
-        self.kms_token = kms_token
+
+    def add_disks_vmware_lso(self):
```
Comment (ebenahar) [inline ocs_ci/ocs/ui/deployment_ui.py:29]: I don't think you need to have this function implemented here. 
1) it is not specific for UI testing
2) it is specific for vsphere, while we already have https://github.com/red-hat-storage/ocs-ci/blob/8dcae7014a94abfc18ef8538f17c94bd98f31f2c/ocs_ci/deployment/deployment.py#L476 which is generic 

Please try to use the call to setup_local_storage() by doing the UI installation after it is called in deploy_ocs_via_operator()

Comment (yitzhak12) [reply ocs_ci/ocs/ui/deployment_ui.py:29]: I think that here we don't need all that is written in the function `setup_local_storage`. We need only the part that Oded added here. However, there is no point in writing a code twice. So my suggestion is to add a new function in the file https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/deployment/helpers/lso_helpers.py - where the function `setup_local_storage` is located. We can call it something like `add_disk_for_vsphere_platform()` and copy-paste the code above to this function. Then we can use it for both in `setup_local_storage` and here.

**Review by yitzhak12 — COMMENTED** [review]
The implementation with selenium looks really nice. Except for my one comment, the PR looks good.

File: ocs_ci/ocs/ui/deployment_ui.py (hunk 2 above)
Comment (OdedViner) [reply ocs_ci/ocs/ui/deployment_ui.py:30]: done

File: ocs_ci/ocs/ui/deployment_ui.py [hunk 3]
```diff
@@ -24,112 +26,42 @@ def __init__(self, driver):
...
+            and config.ENV_DATA.get("platform").lower() == constants.VSPHERE_PLATFORM
+        ):
+            add_disk_for_vsphere_platform()
+
+    def verify_disks_lso_attached(self, timeout=600, sleep=20):
+        """
+        Verify Disks Attached
+
+        timeout (int): Time in seconds to wait
+        sleep (int): Sampling time in seconds
```
Comment (vavuthu) [inline ocs_ci/ocs/ui/deployment_ui.py:46]: please align under Args: section

File: ocs_ci/ocs/ui/deployment_ui.py [hunk 4]
```diff
@@ -243,52 +246,25 @@ def configure_encryption(self):
...
-
-    def verify_ocs_operator_succeeded(self, timeout_install=300, sleep=20):
+    def verify_operator_succeeded(
+        self, operator="OpenShift Container Storage", timeout_install=300, sleep=20
+    ):
         """
-        Verify OCS Installation
+        Verify Operator Installation
 
+        operator (str): type of operator
```
Comment (vavuthu) [inline ocs_ci/ocs/ui/deployment_ui.py:267]: please align under Args section

File: ocs_ci/ocs/ui/deployment_ui.py (hunk 2 above)
Comment (ebenahar) [reply ocs_ci/ocs/ui/deployment_ui.py:30]: do we still need this function? can't we directly call `add_disk_for_vsphere_platform()` from whatever needed under the same condition of LSO?

File: ocs_ci/ocs/ui/deployment_ui.py [hunk 5]
```diff
@@ -292,7 +284,8 @@ def install_ocs_ui(self):
         Install OCS via UI
 
         """
-        self.add_disks_lso()
+        if config.DEPLOYMENT.get("local_storage"):
+            add_disk_for_vsphere_platform()
```
Comment (ebenahar) [inline ocs_ci/ocs/ui/deployment_ui.py:288]: I would add another condition for checking if the platform is vsphere. If not, skip the test

**Review by ebenahar — APPROVED** [review]

**Review by dahorak — APPROVED** [review]

**Pull request merged**